cpp_exercise/src/lib/filehandlerlib.cc

```diff
@@ -0,0 +1,169 @@
+#include "filehandlerlib.h"
+#include "ipcexceptionlib.h"
+
+/**
+ *
+ * @brief FileHandler Constructor for saving the file name
+ * 
+ * 
+ */
+
+FileHandler::FileHandler(const std::string & filename) : _file_name(filename){
+
+}
+
+/**
+ * @brief Destroy the File Handler:: File Handler object and close the file 
+ * 
+ */
+
+FileHandler::~FileHandler(){
+
+    close(_fd);
+    
+}
+
+
+/**
+ * @brief Creating a file
+ * 
+ */
+void FileHandler::createFile(){
+
+    
+   
+    _fd = open(_file_name.c_str(), O_RDWR| O_CREAT | O_TRUNC, _mode);
```

Comment:
I decided to use posix commands becuz I wanted to make all the opening and reading based on posix since I used posix file handling for IPC protocols